processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/EntityImplementationGenerator.java

```diff
@@ -59,14 +62,44 @@ private enum ClassContext {
   private static final String CSV_ROW_NUMBER = "csvRowNumber";
   private final GtfsFileDescriptor fileDescriptor;
   private final GtfsEntityClasses classNames;
+  private final ImmutableMap<String, TypeName> enumIntegerFieldTypes;
 
-  public EntityImplementationGenerator(GtfsFileDescriptor fileDescriptor) {
+  public EntityImplementationGenerator(
+      GtfsFileDescriptor fileDescriptor, List<GtfsEnumDescriptor> enumDescriptors) {
     this.fileDescriptor = fileDescriptor;
     this.classNames = new GtfsEntityClasses(fileDescriptor);
+    this.enumIntegerFieldTypes = createEnumIntegerFieldTypesMap(enumDescriptors);
+  }
+
+  static ImmutableMap<String, TypeName> createEnumIntegerFieldTypesMap(
+      List<GtfsEnumDescriptor> enumDescriptors) {
+    return enumDescriptors.stream()
+        .collect(
+            ImmutableMap.toImmutableMap(
+                GtfsEnumDescriptor::name,
+                EntityImplementationGenerator::chooseEnumIntegerFieldType));
+  }
+
+  static TypeName chooseEnumIntegerFieldType(GtfsEnumDescriptor enumDescriptor) {
+    int minValue = 0;
+    int maxValue = 0;
+    for (GtfsEnumValueDescriptor value : enumDescriptor.values()) {
+      minValue = Math.min(minValue, value.value());
+      maxValue = Math.max(maxValue, value.value());
+    }
+    if (minValue >= Byte.MIN_VALUE && maxValue <= Byte.MAX_VALUE) {
+      return TypeName.BYTE;
+    }
+    if (minValue >= Short.MIN_VALUE && maxValue <= Short.MAX_VALUE) {
+      return TypeName.SHORT;
+    }
+    return TypeName.INT;
   }
 
   private static int lastBitFieldNumber(int fieldCount) {
-    // Each bitField has 32 bits. We need bitField0_ to store 1..32 fields,
+    // All standard GTFS tables have less than 16 fields, so a bitmask fits into short.
```

Comment:
The comment "All standard GTFS" will become obsolete relatively soon, what do you think about replacing it with "Most standard GTFS"?

Reply:
Done.